### PR TITLE
Update Spice lexer

### PR DESCRIPTION
**Pull request by marcauberer**
- Add support for char, short and long literals
- Extend testfile with more examples

### CODE REVIEW

**Comment by marcauberer** [comment]
Failure is probably due to the release of pytest 7.0.0. We seem to run in some [deprecation](https://docs.pytest.org/en/latest/deprecations.html#fspath-argument-for-node-constructors-replaced-with-pathlib-path).

**Comment by birkenfeld** [comment]
Should be fixed now, can you rebase?

**Comment by marcauberer** [comment]
@birkenfeld rebase done. Tests are workling now 👍

File: pygments/lexers/spice.py [hunk 1]
```diff
@@ -44,15 +44,20 @@ class SpiceLexer(RegexLexer):
...
+            (r'(0|[1-9][0-9]*l)', Number.Integer.Long),
             # int_lit
             (r'(0|[1-9][0-9]*)', Number.Integer),
-            # StringLiteral
-            # -- interpreted_string_lit
+            # string_lit
             (r'"(\\\\|\\[^\\]|[^"\\])*"', String),
-            # Tokens
+            # char_lit
+            (r'\'(\\\\|\\[^\\]|[^\'\\])*\'', String.Char),
```
Comment (birkenfeld) [inline pygments/lexers/spice.py:56]: can a char literal really contain more than one character?

Comment (marcauberer) [reply pygments/lexers/spice.py]: Missed that, thanks for pointing that out! Should be fixed now.

**Comment by birkenfeld** [comment]
Thanks!

**Pull request merged**